addons/mail/static/src/core/common/follower_list_view_model.js

```diff
@@ -0,0 +1,73 @@
+import { Record } from "@mail/core/common/record";
+import { rpc } from "@web/core/network/rpc";
+
+export class FollowerListView extends Record {
+    static id = "id";
+    /** @type {Object.<number, import("models").FollowerListView>} */
+    static records = {};
+    /** @returns {import("models").FollowerListView} */
+    static get(data) {
+        return super.get(data);
+    }
+    /**
+     * @param {Data} data
+     * @returns {import("models").FollowerListView|import("models").FollowerListView[]}
+     */
+    static insert(data) {
+        return super.insert(...arguments);
+    }
+
+    /** @type {number} */
+    id;
+    /** @type {number} */
+    thread_id;
+    /** @type {string} */
+    thread_model;
+
+    followers = Record.many("mail.followers");
+
+    /** @type {number} */
+    followersCount;
+
+    selfFollower = Record.one("mail.followers");
+
+    get followersFullyLoaded() {
+        return (
+            this.followersCount ===
+            (this.selfFollower ? this.followers.length + 1 : this.followers.length)
+        );
+    }
+
+    async loadMoreFollowers() {
+        const res = await this.store.env.services.orm.call(
+            this.thread_model,
+            "message_get_followers",
+            [[this.thread_id], 20, this.followers.length]
+        );
+        if (res["mail.followers"]?.length) {
+            this.store.insert(res);
+            this.followers.add(...res["mail.followers"]);
+        }
+    }
+
+    async get_follower() {
+        const res = await rpc("/mail/message/get_followers", {
+            thread_id: this.thread_id,
+            thread_model: this.thread_model,
+            limit: 20,
+            offset: this.followers.length,
+            filter_recipients: false,
+        });
+        const { selfFollower, followersCount, data } = res;
+        const followers = data?.["mail.followers"] || [];
+        if (selfFollower?.["mail.followers"]?.length) {
+            [this.selfFollower] = selfFollower["mail.followers"];
+        }
+        if (followersCount) {
+            this.followersCount = followersCount;
+        }
+        this.followers.add(...followers);
```

Comment:
The usual patter for server returning store data is to `this.store.insert(data)` where `data` is the `Store.get_result()` part. What it does is feed JS global data knowledge with "global" data received from the server.

Then when there are extra contextual data, like `selfFollower` and `followers`, these are plain data identifying records that have been fed with `this.store.insert(data)`. The technique is to retrieve them, e.g. `this.store.Follower.get(selfFollower)`, and then you can do some extra work by linking these records to something else, e.g. here to link with the `FollowerListView` record at hand.

You should not look at `Store.get_result()` content, e.g. how this formatted with `mail.followers` and so on: you should trust it being formatted correctly to feed `this.store.insert()`, and you should rely only on `this.store` functions to retrieve the desired record.